test/integration/controller/job/job_controller_test.go

```diff
@@ -875,15 +876,30 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", ginkgo.Orde
 			return createdProdJob.Spec.Suspend
 		}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(true)))
 
+		originalSelector := maps.Clone(createdProdJob.Spec.Template.Spec.NodeSelector)
+
 		ginkgo.By("creating another localQueue of the same name and in the same namespace as the job")
 		prodLocalQ = testing.MakeLocalQueue(localQueue.Name, ns.Name).ClusterQueue(prodClusterQ.Name).Obj()
 		gomega.Expect(k8sClient.Create(ctx, prodLocalQ)).Should(gomega.Succeed())
 
-		ginkgo.By("job should be unsuspended")
+		ginkgo.By("job should be unsuspended and NodeSelector properly set")
 		gomega.Eventually(func() *bool {
 			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdProdJob)).Should(gomega.Succeed())
 			return createdProdJob.Spec.Suspend
 		}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(false)))
+
+		runningSelector := maps.Clone(createdProdJob.Spec.Template.Spec.NodeSelector)
+		gomega.Expect(runningSelector).NotTo(gomega.Equal(originalSelector))
+
+		expectedSelector := maps.Clone(originalSelector)
+		if expectedSelector == nil {
+			expectedSelector = onDemandFlavor.Spec.NodeLabels
+		} else {
+			for k, v := range onDemandFlavor.Spec.NodeLabels {
+				expectedSelector[k] = v
+			}
+		}
```

Comment:
avoid generating expected values in tests. hard-coding it is more readable https://enterprisecraftsmanship.com/posts/dry-damp-unit-tests/

Reply:
done